PostProcessing/Runtime/Effects/MultiScaleVO.cs

```diff
@@ -180,16 +180,19 @@ public void GenerateAOMap(CommandBuffer cmd, Camera camera, RenderTargetIdentifi
             // Adjust the scaled dimensions based off the dynamic resolution resolution used at the moment.
             m_ScaledWidths[0] = camera.scaledPixelWidth * (RuntimeUtilities.isSinglePassStereoEnabled ? 2 : 1);
             m_ScaledHeights[0] = camera.scaledPixelHeight;
-#endif
-
+#endif            
+            float widthScalingFactor = ScalableBufferManager.widthScaleFactor;
+            float heightScalingFactor = ScalableBufferManager.heightScaleFactor;
             // L1 -> L6 sizes
             for (int i = 1; i < 7; i++)
             {
                 int div = 1 << i;
                 m_Widths[i] = (m_Widths[0] + (div - 1)) / div;
                 m_Heights[i] = (m_Heights[0] + (div - 1)) / div;
-                m_ScaledWidths[i]  = (m_ScaledWidths[0]  + (div - 1)) / div;
-                m_ScaledHeights[i] = (m_ScaledHeights[0] + (div - 1)) / div;
+                // Scaled width and heights have to match their calculations like will happen with dynamic resolution otherwise with odd numbers you can get a difference between what the dynamic resolution version
+                // generates and what we use here.
+                m_ScaledWidths[i] = Mathf.CeilToInt(m_Widths[i] * widthScalingFactor);
```

Comment:
Just comparing the maths to what the Switch does for scaling, Switch clamps the values between 1 and Width or 1 and Height. That's probably just the switch code being overly cautious e.g. the scale factor shouldn't be more than 1, but just double checking that wouldn't be needed here.

Reply:
Good call, I've checked the engine code and the scaling factors are clamped to a max of 1 so we shouldn't need to clamp on the top end but I suppose it's possible this will produce a 0 sized render target in some cases which it sounds like Switch catches, I'm not sure if we catch this case on all platforms and might end up with a 0 sized RT which I don't imagine would produce good results.